source/_posts/2017-01-20-introduction-to-ember-closure-actions.markdown

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "Introduction to Ember Closure Actions"
+date: 2017-01-20 15:07:40 +0100
+comments: true
+author: Jovica Susa
+categories:
+  - Ember
+tags:
+  - ember
+  - javascript
+published: true
+---
+
+Closure actions were introduced in Ember v.1.13.0 and they brought a lot of improvements over old action handling mechanism in Ember. These improvements enabled Ember to adopt new data flow model called -  Data Down Actions Up(DDAU) that simplified communication between parent and child components.
+
+## What are closure acitons?
+
+Closure actions are based on JavaScript closures which are basically functions that remember environment in which they were created. So closure actions are just functions that remember context in which they were defined. Since they are just functions we can pass them as a value and call them directly as a callback. This enables us to pass them to inner components and call them directly from components.
+With the old approach we had to use `sendAction()` from component and call action on controller or route.
+## How to create closure actions?
+
+Every action defined in controller can become a closure action. It’s important to note that it’s possible to make actions defined in routes closure actions but you need to use [ember-route-action-helper](https://github.com/DockYard/ember-route-action-helper) addon for that. Closure actions are created in template using `action` helper which wrapps the action in the current context and returns it as a closure function.
+
+## Example
+
+Let’s define action inside our application controller.
+
+		//controllers/application.js
```

Comment:
@JovicaSusa you can use three backticks ` to open and close code blocks. Also I belive our blog supports language selection so you can probably write something like `'''javascript` (imagine ' are backticks) for opening.